Add routing socket type on macOS (#1867)
nix-rust/nix: CHANGELOG.md

- Added `MntFlags` and `unmount` on all of the BSDs.
- Added `any()` and `all()` to `poll::PollFd`.
  ([#1877](https://github.com/nix-rust/nix/pull/1877))

> - Add `PF_ROUTE` to `SockType` on macOS, all of the BSDs.

Reviewer: ```suggestion
- Add `PF_ROUTE` to `SockType` on macOS and the BSDs.
```

Reviewer: But isn't this still inaccurate?  Because you also added it for iOS, illumos, Haiku, and Fuchsia, I believe.

Author: You are correct, I also moved the entry up the changelog, since these changes have not been released anywhere yet.

Reviewer: I think you forgot to push?

- Added `MntFlags` and `unmount` on all of the BSDs.
- Added `any()` and `all()` to `poll::PollFd`.
  ([#1877](https://github.com/nix-rust/nix/pull/1877))

> - Add `PF_ROUTE` to `SockType` on macOS.

Reviewer: You've got a duped line here.
```suggestion
```

> - Add `PF_ROUTE` to `SockType` on macOS, all of the BSDs and Linux.

Reviewer: This is incorrect, because you _didn't_ add it to Linux.

Author: Ah, you are right. I've updated it now and I do apologize for the sloppy push.
